### ISSUE TITLE
Allow reverting with signature `error Panic(uint)`

### ISSUE THREAD

**Issue by hrkrshnn** [ethereum/solidity#11792]
Currently, solidity doesn't allow defining errors with the following names `Panic` and `Error` (note: name, not signature).

There are use cases where reverting with this error would be useful.

One example is in library fixed point implementation. When adding two fixed point numbers overflows, then it's useful to revert with `Panic(0x11)` to be consistent with how the compiler would revert in such cases. ([List of error codes](https://github.com/ethereum/solidity/blob/develop/libsolutil/ErrorCodes.h), or [Docs](https://docs.soliditylang.org/en/v0.8.7/control-structures.html#panic-via-assert-and-error-via-require))

---

Possible solutions

1. @chriseth suggested allowing something like `import Panic from std.errors`.
2. Allow defining errors with signature `Panic(uint)` and `Error(string)`.
3. Use inline assembly to mimic the error: no need for the compiler to do anything
4. Use a collision hack. Find `<name>(uint)` with the same hash as `Panic(uint)` and similarly for `Error(string)`.

**Comment by hrkrshnn** [comment]
Pinging @paulrberg since this could be useful for https://github.com/hifi-finance/prb-math/blob/main/contracts/PRBMath.sol

**Comment by PaulRBerg** [comment]
Yes, this would be helpful. [This](https://github.com/hifi-finance/prb-math/blob/v2.2.0/contracts/PRBMathUD60x18Typed.sol#L39-L41) is the specific place where I am reverting with a custom error that mimics `Panic`:

```sol
if (rValue < x.value) {
    revert PRBMathUD60x18__AddOverflow(x.value, y.value);
}
```

If I could, I would switch to the `Panic` code instead of using my custom defined error.

**Comment by hrkrshnn** [comment]
Decision: do 1 and 2 at the same time.

**Comment by Amxx** [comment]
Any news on that feature?

**Comment by cameel** [comment]
Not yet but this is one of the features we have in the priority backlog so it should be one of the first to pick up once we have some more free cycles.

(1) Looks like it depends on stdlib though and that's not ready yet.

**Comment by Amxx** [comment]
Again, any news on that feature?
